wordpress-mobile/WordPress-iOS
# Added a working Classic / Gutenberg switch option in the editing menu.

### Description:

Closes #10554.

This PR adds an option to the "More Menu" ("..." in the nav bar) to switch between Aztec and Gutenberg.

Non-saved edits should be maintained across editors.

### Details:

This PR also removes [some old restoration code](https://github.com/wordpress-mobile/WordPress-iOS/pull/10565/files#diff-705f087506da53949ae67cbac0ed8d98L3259) which wasn't really being used, since it would've needed to be updated for these changes.

### Demo:

![switcheditors](https://user-images.githubusercontent.com/1836005/49189943-08c69c80-f34f-11e8-93f7-5178db4818d3.gif)

### Testing:

**Test 1:**

Make sure the switch option does not show up if Gutenberg is disabled.

**Test 2:**

Enable Gutenberg and:

- Make sure the switch option shows up in Gutenberg for all posts.
- Make sure the switch option shows up in Aztec for posts that contain Gutenberg blocks.

**Test 3:**

Try opening a post in any editor, make some edits and switch to the other editor.  Make sure the edits are kept.

### Limitations:

Due to the complexity of recognizing when a post has changes, I'm leaving change-tracking outside of the scope of this PR.

This means that the "This post has changes" alert will be worked on separately.

This should not be a problem since the feature is internal and it won't affect the logic for users that don't have Gutenberg.
